util/test/Test/Pos/Util/Golden.hs

```diff
@@ -35,6 +37,27 @@ embedGoldenTest :: FilePath -> ExpQ
 embedGoldenTest path =
     makeRelativeToTestDir ("golden/" <> path) >>= embedStringFile
 
+-- | Test if prettified JSON and unformatted JSON are equivalent.
+goldenFileEquiv :: (Eq a)
+                => Either String a -> Either String a -> Either String Bool
+goldenFileEquiv prettified unformatted = do
+        case prettified of
+            Left err -> Left $ "could not decode: " <> show err
+            Right x ->  case unformatted of
+                            Left err -> Left $ "could not decode: " <> show err
+                            Right x' -> Right $ x' == x
```

Comment:
Here we could leverage the `Either` Monad, and write something like
```
p <- prettified
u <- unformatted
pure (p == u)
```
However if you want to add the `"could not decode: "` to the error message, you'd have to add something like.
```
... = do
    p <- prefixError prettified
    u <- prefixError unformatted
    pure (p == u)
  where
    prefixError (Left err) = Left $ "could not decode: " <> err
    prefixError r@(Right _) = r
```
In the end, that looks about the same as what you wrote. So it's a judgment call here I think 🙂

Reply:
Actually, since you add the prefix at usage sites (e.g. [here](https://github.com/input-output-hk/cardano-sl/pull/3746/files#diff-14fd7e44f26401fbd532a9c10d6e60f0R86)), maybe it can be omitted in this function, for the nice & sleek Monad style.

Reply:
Since this function is actually comparing "either successfully decoded values, or decode error messages", perhaps we could rename from `goldenFileEquiv` to `goldenValueEquiv`.

Reply:
Glossed over this, thanks for pointing it out!